OSError on macOS 12.2 running on arm

Setup: macOS 12.1, oscrypto 1.2.1, Python 3.9.5

Not sure where to even start to look for this problem. Running the newest mac version on arm and not Intel. The OSStatus 62385568 did not give me any clues.

```
/Users/xxx/.local/share/virtualenvs/QpmVhfBc/lib/python3.9/site-packages/oscrypto/_mac/asymmetric.py:1027: in rsa_pkcs1v15_encrypt
    key_length = certificate_or_public_key.byte_size
/Users/xxx/.local/share/virtualenvs/QpmVhfBc/lib/python3.9/site-packages/oscrypto/_asymmetric.py:225: in byte_size
    return self.public_key.byte_size
/Users/xxx/.local/share/virtualenvs/QpmVhfBc/lib/python3.9/site-packages/oscrypto/_mac/asymmetric.py:255: in public_key
    handle_sec_error(res)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

error = 62385568, exception_class = <class 'OSError'>
    
>       raise exception_class(output)
E       OSError: OSStatus 62385568

/Users/xxx/.local/share/virtualenvs/QpmVhfBc/lib/python3.9/site-packages/oscrypto/_mac/_security.py:57: OSError
```

--- a/oscrypto/_mac/_security_cffi.py
+++ b/oscrypto/_mac/_security_cffi.py
@@ -103,6 +103,7 @@ ffi.cdef("""
     SecTransformRef SecSignTransformCreate(SecKeyRef key, CFErrorRef *error);
     SecCertificateRef SecCertificateCreateWithData(CFAllocatorRef allocator, CFDataRef data);
     OSStatus SecCertificateCopyPublicKey(SecCertificateRef certificate, SecKeyRef *key);
+    SecKeyRef SecCertificateCopyKey(SecCertificateRef certificate);
     CFStringRef SecCopyErrorMessageString(OSStatus status, void *reserved);
     OSStatus SecTrustCopyAnchorCertificates(CFArrayRef *anchors);
     CFDataRef SecCertificateCopyData(SecCertificateRef certificate);

--- a/oscrypto/_mac/_security_ctypes.py
+++ b/oscrypto/_mac/_security_ctypes.py
@@ -186,6 +186,11 @@ try:
     ]
     Security.SecCertificateCreateWithData.restype = SecCertificateRef
 
+    Security.SecCertificateCopyKey.argtypes = [
+        SecCertificateRef,
+    ]
+    Security.SecCertificateCopyKey.restype = SecKeyRef
+
     Security.SecCertificateCopyPublicKey.argtypes = [
         SecCertificateRef,
         POINTER(SecKeyRef)

--- a/oscrypto/_mac/asymmetric.py
+++ b/oscrypto/_mac/asymmetric.py
@@ -261,9 +261,12 @@ class Certificate(_CertificateBase):
                 sec_cert_ref = self.sec_certificate_ref
 
             sec_public_key_ref_pointer = new(Security, 'SecKeyRef *')
-            res = Security.SecCertificateCopyPublicKey(sec_cert_ref, sec_public_key_ref_pointer)
-            handle_sec_error(res)
-            sec_public_key_ref = unwrap(sec_public_key_ref_pointer)
+            if osx_version_info >= (10, 14):
+                sec_public_key_ref = Security.SecCertificateCopyKey(sec_cert_ref)
+            else:
+                res = Security.SecCertificateCopyPublicKey(sec_cert_ref, sec_public_key_ref_pointer)
+                handle_sec_error(res)
+                sec_public_key_ref = unwrap(sec_public_key_ref_pointer)
             self._public_key = PublicKey(sec_public_key_ref, self.asn1['tbs_certificate']['subject_public_key_info'])
 
         return self._public_key

